common/lib/xmodule/xmodule/library_tools.py

```diff
@@ -16,16 +16,21 @@ class LibraryToolsService(object):
     def __init__(self, modulestore):
         self.store = modulestore
 
-    def _get_library(self, library_key):
+    def _get_library(self, library_key, version=None):
         """
         Given a library key like "library-v1:ProblemX+PR0B", return the
         'library' XBlock with meta-information about the library.
 
+        A specific version may be specified.
+
         Returns None on error.
         """
         if not isinstance(library_key, LibraryLocator):
             library_key = LibraryLocator.from_string(library_key)
-        assert library_key.version_guid is None
+
+        library_key = LibraryLocator(
+            org=library_key.org, library=library_key.library, branch=library_key.branch, version_guid=version
+        )
```

Comment:
I don't think `version` should be a separate argument here, because there's potentially already a version value embedded in the first argument (library_key) as well.

You can remove the few lines above and just call `_get_library(library_key.for_version(version))` to get the same effect. (Or `_get_library(LibraryLocator.from_string(library_key_str).for_version(version))`)